### PR TITLE
lsp-types patch for platform independence

### LINKED ISSUE #10117: Package ids are platform dependent
Currently, our package ids are platform dependent since they end up embedding file paths in the resulting dalf. On Windows we normalize those to `\` whereas on linux we normalize to `/` so they are clearly different.

The fix for this would probably need to be here https://github.com/haskell/lsp/blob/9eb60f9fefa88fbb98d8c6f3eb998292c88c05ad/haskell-lsp-types/src/Language/Haskell/LSP/Types/Uri.hs#L197.

Note that this needs careful testing to make sure we don’t break vscode. I believe it will work but this has been a nightmare in the past.

### PR DESCRIPTION
**Pull request by robin-da**
Fixes #10117.

This patches the lsp-types hackage dependency to make sure that only POSIX encoded filepaths end up in the Daml-lf package.
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Nice work, thank you!

File: bazel-haskell-deps.bzl [hunk 1]
```diff
@@ -71,6 +72,52 @@ haskell_binary(
...
+    http_archive(
+        name = "lsp-types",
+        build_file_content = """
+load("@rules_haskell//haskell:cabal.bzl", "haskell_cabal_library")
+load("@stackage//:packages.bzl", "packages")
+haskell_cabal_library(
+    name = "lsp-types",
+    version = "{version}",
+    srcs = glob(["**"]),
+    deps = [
```
[the remark this answers is not in the document]
Comment (robin-da) [reply bazel-haskell-deps.bzl:84]: cool, I'll try that.

File: bazel_tools/lsp-types-normalisation.patch [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
++++ b/src/Language/LSP/Types/Uri.hs
+@@ -110,12 +110,12 @@ platformAdjustFromUriPath systemOS authority srcPath =
+     else let
+       firstSegment:rest = (FPP.splitDirectories . tail) srcPath  -- Drop leading '/' for absolute Windows paths
+       drive = if FPW.isDrive firstSegment
+-              then FPW.addTrailingPathSeparator firstSegment
++              then FPP.addTrailingPathSeparator firstSegment
+               else firstSegment
+-      in FPW.joinDrive drive $ FPW.joinPath rest
++      in FPP.joinDrive drive $ FPP.joinPath rest
```
[the remark this answers is not in the document]
Comment (robin-da) [reply bazel_tools/lsp-types-normalisation.patch:13]: FPW -> FPP (i.e. use posix directory filepath manipulation functions here).

File: bazel_tools/lsp-types-normalisation.patch [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+-filePathToUri = (platformAwareFilePathToUri System.Info.os) . FP.normalise
++filePathToUri = (platformAwareFilePathToUri System.Info.os) . normalise'
+ 
+ {-# WARNING platformAwareFilePathToUri "This function is considered private. Use normalizedUriToFilePath instead." #-}
+ platformAwareFilePathToUri :: SystemOS -> FilePath -> Uri
+@@ -187,10 +187,21 @@ instance Hashable NormalizedFilePath where
+ instance IsString NormalizedFilePath where
+     fromString = toNormalizedFilePath
+ 
++normalise' :: FilePath -> FilePath
```
[the remark this answers is not in the document]
Comment (robin-da) [reply bazel_tools/lsp-types-normalisation.patch:25]: added a comment.

File: compiler/damlc/tests/daml-test-files/PlatformIndependence.daml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+module PlatformIndependence where
```
[the remark this answers is not in the document]
Comment (robin-da) [reply compiler/damlc/tests/daml-test-files/PlatformIndependence.daml:1]: I'll add this in a follow up PR.

**Pull request merged**